metadata/buckets.go

```diff
@@ -182,6 +182,41 @@ func createBucketIfNotExists(tx *bolt.Tx, keys ...[]byte) (*bolt.Bucket, error)
 	return bkt, nil
 }
 
+func namespacesBucketPath() []byte {
+	return bucketKeyVersion
+}
+
+func getNamespacesBucket(tx *bolt.Tx) *bolt.Bucket {
+	return getBucket(tx, namespacesBucketPath())
+}
+
+func getSharableBucket(tx *bolt.Tx, dgst digest.Digest) *bolt.Bucket {
```

Comment:
It is a bit scary to see we have O(N) (N = # of namespaces x # of labels) search here.

We have discussed label indexing before (#2980), but haven't implemented yet.

Reply:
I am not sure of other way to do it if the label is on the namespace we want to share *from* but I agree it could be a potential problem. I wonder what the best way to measure this impact might be? Will try to think about it.

Reply:
Just curious why O(N) will be problematic? O(N) algorithm generally perform OK theoretically. 
In addition, for the label indexing, do you expect sth like a hashmap structure?